src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/UpdaterFactory.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater
+{
+    public class UpdaterFactory
+    {
+        public static PackageUpdater? GetPackageUpdater(string path, ILogger logger)
+        {
+            try
+            {
+                XDocument doc = XDocument.Load(path);
+                PackageUpdater packageUpdater = new PackageUpdater(doc, logger);
+                return packageUpdater;
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error loading project file {path}");
+                return null;
+            }
+        }
+
+        public static ConfigUpdater? GetConfigUpdater(string path, ILogger logger)
+        {
+            try
+            {
+                XDocument doc = XDocument.Load(path);
+                ConfigUpdater configUpdater = new ConfigUpdater(doc, logger);
+                return configUpdater;
+            }
+            catch (Exception e)
```

Comment:
As in other places, if we can catch a more specific exception, that would be good. It will avoid accidentally catching some unexpected catastrophic event that really ought to not be handled.

Reply:
I added a lot of try/catch for the UpdateRunner + UpdateFactory and was just generally concerned about unexpected errors happening and I thought it would be better to catch it as early as possible instead of letting it bubble up? I am wondering what would be the best practice for the error handling in this case?